services/motion/builtin/builtin_test.go

```diff
@@ -570,30 +562,16 @@ func TestMoveOnGlobe(t *testing.T) {
 
 	t.Run("go around an obstacle", func(t *testing.T) {
 		t.Parallel()
-		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, dst, nil)
+		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil)
 		motionCfg := &motion.MotionConfiguration{PositionPollingFreqHz: 4, ObstaclePollingFreqHz: 1, PlanDeviationMM: epsilonMM}
 
 		boxPose := spatialmath.NewPoseFromPoint(r3.Vector{50, 0, 0})
 		boxDims := r3.Vector{5, 50, 10}
 		geometries, err := spatialmath.NewBox(boxPose, boxDims, "wall")
 		test.That(t, err, test.ShouldBeNil)
 		geoObstacle := spatialmath.NewGeoObstacle(gpsPoint, []spatialmath.Geometry{geometries})
-
-		moveRequest, err := ms.(*builtIn).newMoveOnGlobeRequest(
-			ctx,
-			fakeBase.Name(),
-			dst,
-			injectedMovementSensor.Name(),
-			[]*spatialmath.GeoObstacle{geoObstacle},
-			motionCfg,
-			extra,
-		)
+		startPose, err := fakeBase.CurrentPosition(ctx)
 		test.That(t, err, test.ShouldBeNil)
-		waypoints, err := moveRequest.plan(ctx)
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, len(waypoints), test.ShouldBeGreaterThan, 2)
```

Comment:
There's not really a need to run the exact same planning query twice. It should be sufficient to check `CurrentPosition` after the `MoveOnGlobe` call.

Reply:
Checking the length of the plan is the way that we can determine obstacles are being avoided.  I would rather leave this in

Reply:
The length of the plan and whether or not the base is traveling in a straight line are no longer particularly related with PTGs. So this will need to be updated regardless in order to adequately test anything.

The following test, where obstacles result in an inability to plan, should cover the same general thing.

Reply:
I could try to add a CheckPlan call to this

Reply:
CheckPlan would be great here actually

Reply:
Took a stab at adding CheckPlan. The different things needed for CheckPlan that don't exist here (frame system, etc) are kind of a giant pain to put together and would require re-implementing waaaay to much code from elsewhere in the motion stack. I've put back the plan and the length check, but CheckPlan isn't easy to add here at the moment.